velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -268,6 +268,19 @@ class PrestoSerializerTest
     serde_->serializeEncoded(rowVector, &arena, &paramOptions, &out);
   }
 
+  void assertEqualEncoding(const RowVector* expected, const RowVector* actual) {
+    for (auto i = 0; i < expected->childrenSize(); ++i) {
+      VELOX_CHECK_EQ(
+          actual->childAt(i)->encoding(), expected->childAt(i)->encoding());
+
+      if (expected->childAt(i)->encoding() == VectorEncoding::Simple::ROW) {
```

Comment:
Do we allow nested encodings? Dict(Dict(Dict(Flat)))

I'm not sure if it makes sense to allow these. If we do, then let's make sure we have test coverage. Otherwise, let's document the limitation and add checks to the code.

Also, do we allow flat Arrays with dictionary-encoded elements? Similar question for maps.

Reply:
Only rowVector children will preserve encoding today; added a comment to clarify